ssas/service/public/api.go

```diff
@@ -39,12 +39,69 @@ type RegistrationRequest struct {
 }
 
 type MFARequest struct {
-	CmsID       string  `json:"cms_id"`
+	CMSID       string  `json:"cms_id"`
 	FactorType  string  `json:"factor_type"`
 	Passcode    *string `json:"passcode,omitempty"`
 	Transaction *string `json:"transaction,omitempty"`
 }
 
+type PasswordRequest struct {
+	CMSID    string `json:"cms_id"`
+	Password string `json:"password"`
+}
+
+/*
+	VerifyPassword is mounted at POST /authn and responds with the account status for a verified username/password
+ 	combination.
+*/
+func VerifyPassword(w http.ResponseWriter, r *http.Request) {
+	var (
+		err        error
+		trackingID string
+		passReq    PasswordRequest
+	)
+
+	setHeaders(w)
+
+	bodyStr, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		jsonError(w, "invalid_client_metadata", "Request body cannot be read")
+		return
+	}
+
+	err = json.Unmarshal(bodyStr, &passReq)
+	if err != nil {
+		service.LogEntrySetField(r, "bodyStr", "<redacted>")
+		jsonError(w, "invalid_client_metadata", "Request body cannot be parsed")
+		return
+	}
+
+	trackingID = uuid.NewRandom().String()
+	event := ssas.Event{Op: "VerifyOktaPassword", TrackingID: trackingID, Help: "calling from public.VerifyPassword()"}
+	ssas.OperationCalled(event)
+	passwordResponse, err := GetProvider().VerifyPassword(passReq.CMSID, passReq.Password, trackingID)
+	if err != nil {
+		jsonError(w, "invalid_client_metadata", err.Error())
+		return
+	}
+
+	body, err := json.Marshal(passwordResponse)
```

Comment:
The password response just tells the user to `proceed to MFA verification`. The next step, authn/request, to do the mfa challenge requires a `factor_type`, but as a consumer of this api, I have no idea what the factor_types are.

Reply:
Presumably, you know what factor type you registered with Okta, and originally we were keeping our cards close to our chest.  Now that we have password authentication completed before this endpoint is reached, we can give them a list in `POST /authn`.

Reply:
BCDA-1624 has been created to remind us of this enhancement